### PR TITLE
Info button and hints

### LINKED ISSUE #1913: Add clarifications / details for quests
There are many situations where I am/was not sure how to answer.
Often it was possible to find an answer here on github or in the OSM wiki, sometimes not.
Since I see I am not the only one who sometimes doesn't know how to answer (example: #1889), I think it would be beneficial to have answers for relatively common situations provided directly in the app, behind some sort of "details" button (or maybe a link to the corresponding page in the OSM wiki, but often this is not particularly helpful).
Some of the questions/problems might not have a clear answer, but  for those that have this "details" button could help.

I gathered a bunch of examples where I was not sure and looked up the answer, or where I still don't know how to handle it correctly.
For some of them the answer seems rather obvious to me now, but initially it wasn't.
* lit (mostly when light is coming from nearby streets)
	* how bright does it need to be to be count as lit: see other people? be able to spot dog droppings on the ground? read a plan?
	* does it have to be lit full width?
	* when does a sports field neet to be lit to answer `yes`?
* surfaces
	* ~difference between paving stones and sett~ (better with the updated images)
	* what surfaces are tiles / natural stones (#3189, #616)
	* how precise should I be when splitting a path?
		* e.g. 2 m sett driveway on asphalt pavement?
		* e.g. 30 cm concrete speed bump on asphalt road?
		* what about crossing streets/paths with different surfaces?
	* different surfaces across the road
		* ~e.g. one lane concrete, one lane asphalt~
		* ~e.g. asphalt with paving stone patterns~
		* should just be _paved_, which forces adding a note (which is rather tedious for long roads with many segments)
* building type
	* detached house: simply defined by no contact to other houses?
	* semi-detached: should the halves be similar? What about top/bottom separation ([#2685](https://github.com/streetcomplete/StreetComplete/issues/2685#issuecomment-809011489))
	* ~apartment building: what is the difference to a house?~ (_single family_ in description should be sufficient)
	* "changed" buildings
		* e.g. old apartment building containing only offices
		* e.g. old industrial building containing apartments
		* there were issues about that, but it's not mentioned in the app
	* what kind of building is
		* container, driveway to underground parking, staircase, elevator, "cage with roof and walls" (often containing garbage/recycling containers or bicycle parking)
		* bus stop shelter: public transport or roof? (probably should be amenity=shelter)
		* subway/metro: public transport or train station?
* building levels
	* what is the lowest level (unclear semi-undergroud levels)
		* windows above ground, but no entrance
		* door with immediate steps down (steps outside or inside the building)
		* driveway/ramp to a garage in the cellar
	* how do levels on the roof count
		* e.g. flat roof with some small "building"
	* what if one part has x levels and 1 in the roof, other part has x+1 levels and none in the roof?
	* when does a level count as "in the roof"?
		* especially unclear for flat roofs where the top level is often smaller than the others
* roof shape
	* what to use for "half" roofs (half gabled exists as skillion, others have no equivalent)
	* what if the two ends are different? common for gabled and hipped
	* gabled roof at a corner: is this already hipped?
	* for questions/discussions: having names of the shapes would help!
* name of place
	* do generic names like _information_, _butcher_, _exchange_ count as names?
	* should "type names" be included? The type-name combination often feels like the "correct" name for small shops, e.g. "Hair Salon Herbert", or "Cosmetics Anna"
* wheelchair
	* when is a kiosk accessible? Doe it depend on height of counter?
	* is there some minimum door width?
	* height limits for that one step for partial access? OSM wiki mentions 7 cm
	* what if there is a button/doorbell to get assistance?
	* no direct obstacle, but steep paths with bad surfaces (grass, compacted,...)
	* the current descrition is focused on indoor. Answering _no_ even if a reastaurant has accessible outdoor seating is corect according to description, but feels wrong
	* what if the feature is accessible, but the path is not (e.g toilets with wheelchair sign, but the pavement is not wheelchair-compatible)
* pavement
	* what counts as a pavement? The intuitive understanding seems to vary across languages (even for sidewalk and pavement I would answer differently)
	* multi-lane roads: should I add the pavement for each lane, or only the closest one?
	* often there are roads "embedded" in a square - there is no pavement directly at those roads, and it is not mapped separately, but the square has pavements.
	* pavement separated from the road by grass or trees: does it still count as pavement of that road?
	* separate path on one side only, but pavement on both sides - what to answer? (I think change if the sidewalk quest for this is already planned)
* tactile pavings: is it really `no` if there is a tactile paving along the crossing, but not at the end?
* bridges: names and a short description of each type would sometimes be very helpful, as many bridges seem to be "between" two of the provided images (often a very slight arch only, almost like a beam bridge)
* `who is allowed to park here`: what if only specific vehicles, e.g. buses?

I don't need the answers here, this is only about whether such clarifications should be added to the app... at least the ones that fit in one or two sentences.

### PR DESCRIPTION
**Pull request by westnordost**
Kind of fixes #1913. Many suggestions in #1913 are not actionable because the definition of certain tags is vague in the wiki and I cannot simply make something up. (E.g. `lit` status on bus stops.)

This PR mostly moves the hints into an area that is expanded when pressing the info button that is shown on the right side of the question, however, adds a few hints where I found they make sense. I did not move all texts that appear as hints into the info area because with this implemented, we now have three places in which things can be clarified and each are for different use cases.

<table>
  <tr><th>Tool</th><th>Use case</th><th>Example</th></tr>
  <tr>
    <td>Info hint text & images</td>
    <td>Doubts on the question itself (terms used, etc.), <b>how to</b> answer or how to find the answer.</td>
    <td>Explanation of what is halal, what is tactile paving, what is a sanitary dump station, when a level counts as a roof level and when as a normal level etc., explanation of the street side puzzle UI</td>
  </tr>
  <tr>
    <td>Answer option (including in "uh...")</td>
    <td>Doubts on how to answer a specific situation.</td>
    <td>Buildings with several uses, e.g. retail and residential</td>
  </tr>
  <tr>
    <td>Label text for an input field</td>
    <td>Instruction on how to answer, i.e. in what format the answer is expected. These should always be shown and be short.</td>
    <td>"Pick one"; that road names should not be abbreviated; that on the vegetarian quest one should only answer "yes" if there are proper options...</td>
  </tr>
</table>

### CODE REVIEW

**Comment by westnordost** [comment]
If anyone wants to review this, the most interesting part is the strings.xml. I added some strings and modified some strings.

**Comment by westnordost** [comment]
By the way I didn't add an animation because it is somewhat cumbersome to do. Since sooner or later the UI is redone in Jetpack Compose, I didn't want to put an unreasonable amount of effort in it:
- `animateLayoutChanges`=true seems to be bugged when used with bottom sheets. The animation looked fine, but it also made the bottom sheet form go to the top of the screen (wth)
- a normal object animator animation with `scaleY` will literally just scale the content, which looks odd. Animating the layout height on the other hand had the effect that the layout children tried to scale themselves to fit the mandated height during animation (i.e. images weirdly grow in size)

**Comment by FloEdelmann** [comment]
Had a look at the code and strings now, they look fine!

**Comment by FloEdelmann** [comment]
Should we add the table from the PR description to some more permanent place, like the quest docs?

File: app/src/main/java/de/westnordost/streetcomplete/quests/foot/ProhibitedForPedestriansAnswer.kt [hunk 1]
```diff
@@ -3,6 +3,5 @@ package de.westnordost.streetcomplete.quests.foot
 enum class ProhibitedForPedestriansAnswer {
     YES,
     NO,
-    IS_LIVING_STREET,
```
Comment (FloEdelmann) [inline app/src/main/java/de/westnordost/streetcomplete/quests/foot/ProhibitedForPedestriansAnswer.kt:6]: I guess this was removed on purpose? Now a test is failing, see https://github.com/streetcomplete/StreetComplete/actions/runs/8960282103/job/24606606961:

> app/src/test/java/de/westnordost/streetcomplete/quests/foot/AddProhibitedForPedestriansTest.kt:57:39 Unresolved reference: IS_LIVING_STREET

Comment (westnordost) [reply app/src/main/java/de/westnordost/streetcomplete/quests/foot/ProhibitedForPedestriansAnswer.kt:6]: oops. Thanks!

**Comment by westnordost** [comment]
> Should we copy the table from the PR description to some more permanent place, like the quest docs?

To "contributing a quest?". I don't know, I don't really maintain this, that's @matkoniecz project.

File: app/src/main/java/de/westnordost/streetcomplete/quests/foot/ProhibitedForPedestriansAnswer.kt (hunk 1 above)
Comment (FloEdelmann) [reply app/src/main/java/de/westnordost/streetcomplete/quests/foot/ProhibitedForPedestriansAnswer.kt:6]: Fixed in https://github.com/streetcomplete/StreetComplete/commit/1fe1f79925c1b72592212f64d5b31609a03ac288

**Comment by Helium314** [comment]
Finally had time for a look. Very nice, thank you!

**Comment by mnalis** [comment]
Can somebody check if the info button is visible for them in **dark mode**?

small_Screenshot_20240523_085403_StreetComplete Dev small_Screenshot_20240523_084956_StreetComplete Dev

(I've merged the latest master to my fork for some testing, so I cannot easily say if it is due to my local changes until it is released)

**Comment by westnordost** [comment]
Right, fixed it.

**Comment by HolgerJeromin** [comment]
ref 7a40b562cc6251a03eda951b6319cd5c170243ea

**Pull request merged**